### DISCUSSION TITLE
1.6.2

### DISCUSSION THREAD

**Discussion by kirb** [LegacyUpdate/LegacyUpdate#105] [Announcements]
This release will feel very minor on the surface, but I put a bunch of work into improving the codebase.

* Switches the file downloader library to one that seems to be a lot more stable, and reliable in every condition - slow internet, insufficient RAM, internet faster than your RAM/hard drive can keep up with, etc. It is also able to reconnect and continue if your connection drops during a download.
* Refactors the ActiveX control from the hefty Microsoft MFC framework to the lighter-weight ATL framework. This brings down the file size of the DLL.
* Uses more efficient compression on the installer (LZMA rather than Gzip), bringing the download size down even further.
* Improves error handling in the ActiveX control, so that in the unlikely case of the Legacy Update ActiveX control itself failing to do something, more useful error information will be displayed.
* Fixes installing Service Packs on Windows Vista/7 again - this was supposed to be fixed in 1.6.1 but I created another bug in the process.
* Fixes the LegacyUpdateTemp user lacking administrator permission on non-English installs of Windows, causing a "log on as an administrator" error after Legacy Update setup automatic restarts.

The massive changes in this release seem to have tripped up the exceptions Legacy Update was receiving from antivirus vendors. [Currently](https://www.virustotal.com/gui/file/956866a9772ac07ee05e6f7cb00ae2f93d9206ea6bb992f532184e6d223208ec/detection), ~~Microsoft~~ and BitDefender are triggering false positives. This should be corrected soon, once both companies respond to my tickets. (Update: Microsoft Defender false positive resolved.)

<hr /><em>This discussion was created from the release <a href='https://github.com/kirb/LegacyUpdate/releases/tag/v1.6.2.0'>1.6.2</a>.</em>

**Comment by chaowb0936** [comment]
Can we add Windows 8/8.1 update support to "https://legacyupdate.net/"?

**Reply by kirb** [reply]
I didn’t enable the option to install the ActiveX control for Vista and up because it’s not needed, and because the site originally didn’t work in IE9+. I worked around that issue, but some people have asked to have the web option anyway, so I probably will make it an option.
